src/main/java/teammates/common/datatransfer/FeedbackSessionAttributes.java

```diff
@@ -316,6 +316,24 @@ public boolean isClosingWithinTimeLimit(int hours) {
                && differenceBetweenDeadlineAndNow >= hours - 1
                && differenceBetweenDeadlineAndNow < hours;
     }
+    
+    /**
+     * returns true if the session is closed within the past hour of calling this function
+     */
+    public boolean isClosedWithinPastHour() {
+        Calendar now = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        // Fix the time zone accordingly
+        now.add(Calendar.MILLISECOND, (int) (60 * 60 * 1000 * timeZone));
+
+        Calendar deadline = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        deadline.setTime(endTime);
+        deadline.add(Calendar.MINUTE, gracePeriod);
+
+        long nowMillis = now.getTimeInMillis();
+        long deadlineMillis = deadline.getTimeInMillis();
+        long differenceBetweenNowAndDeadline = (deadlineMillis - nowMillis) / (60 * 60 * 1000);
+        return differenceBetweenNowAndDeadline == 0 && deadlineMillis < nowMillis;
+    }
```

Comment:
Improvements to this method can be done as a separate PR, as long as you do it immediately after this one.